server/schedulers/split_bucket.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schedulers
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/tikv/pd/pkg/syncutil"
+	"github.com/tikv/pd/server/core"
+	"github.com/tikv/pd/server/schedule"
+	"github.com/tikv/pd/server/schedule/operator"
+	"github.com/tikv/pd/server/statistics/buckets"
+	"github.com/tikv/pd/server/storage/endpoint"
+	"github.com/unrolled/render"
+)
+
+const (
+	// SplitBucketName is the split bucket name.
+	SplitBucketName = "split-bucket-scheduler"
+	// SplitBucketType is the spilt bucket type.
+	SplitBucketType = "split-bucket"
+	// defaultHotDegree is the default hot region threshold.
+	defaultHotDegree = 3
+)
+
+func init() {
+	schedule.RegisterSliceDecoderBuilder(SplitBucketType, func(args []string) schedule.ConfigDecoder {
+		return func(v interface{}) error {
+			return nil
+		}
+	})
+
+	schedule.RegisterScheduler(SplitBucketType, func(opController *schedule.OperatorController, storage endpoint.ConfigStorage, decoder schedule.ConfigDecoder) (schedule.Scheduler, error) {
+		conf := initSplitBucketConfig()
+		if err := decoder(conf); err != nil {
+			return nil, err
+		}
+		conf.storage = storage
+		return newSplitBucketScheduler(opController, conf), nil
+	})
+}
+
+func initSplitBucketConfig() *splitBucketSchedulerConfig {
+	return &splitBucketSchedulerConfig{
+		Degree: defaultHotDegree,
+	}
+}
+
+type splitBucketSchedulerConfig struct {
+	mu      syncutil.RWMutex
+	storage endpoint.ConfigStorage
+	Degree  int `json:"degree"`
```

Comment:
Why not keep using HotDegree?

Reply:
the function of statistics is different.